src/iotconnect_event.c

```diff
@@ -262,33 +262,48 @@ static char *create_ack(
         if (!cJSON_AddNumberToObject(ack_data, "st", to_ack_status(success, message_type))) goto cleanup;
     }
 
-    result = cJSON_PrintUnformatted(ack_json);
+    result = (const char *) cJSON_PrintUnformatted(ack_json);
 
     // fall through
     cleanup:
     cJSON_Delete(ack_json);
     return result;
 }
 
-char *iotcl_create_ack_string_and_destroy_event(
+const char *iotcl_create_ack_string_and_destroy_event(
         IotclEventData data,
         bool success,
         const char *message
 ) {
     if (!data) return NULL;
     // already checked that ack ID is valid in the messages
     char *ack_id = cJSON_GetObjectItemCaseSensitive(data->data, "ackId")->valuestring;
-    char *ret = create_ack(success, message, data->type, ack_id);
+    const char *ret = create_ack(success, message, data->type, ack_id);
```

Comment:
create_* functions should return char *, not const char* because they create by making a copy that the user is free to do what they want with. All create* function resuts must be freed, which implies that they are modifiable values.  This could be a semantic thing that I am clinging on to, but I am open to suggestions.